### PR TITLE
Berinstein-Vazirani's algorithm.

### PR DESCRIPTION
**Pull request by joelin0**
To resolve Issue #37.

### CODE REVIEW

File: grove/bernstein_vazirani/bernstein_vazirani.py [hunk 1]
```diff
@@ -0,0 +1,170 @@
...
+    p.inst(X(ancilla), H(ancilla))
+
+    # Apply Hadamard, Unitary function, and Hadamard again
+    p.inst(map(H, qubits))
+    p += oracle
+    p.inst(map(H, qubits))
+    return p
+
+
+def run(cxn, oracle, qubits, ancilla):
```
Comment (tarballs-are-good) [inline grove/bernstein_vazirani/bernstein_vazirani.py]: call this `run_bernstein_vazirani`

File: grove/bernstein_vazirani/bernstein_vazirani.py [hunk 2]
```diff
@@ -0,0 +1,170 @@
...
+    """
+    Runs the Bernstein-Vazirani algorithm.
+
+    Given a QVM connection, an oracle, the input bits, and ancilla,
+    find the :math:`\\mathbf{a}` and :math:`b` corresponding to the function
+    represented by the oracle.
+
+    :param SyncConnection cxn: the QVM connection to use to run the programs
```
Comment (tarballs-are-good) [inline grove/bernstein_vazirani/bernstein_vazirani.py]: Does this need to be a SyncConnection?

**Comment by joelin0** [comment]
Should be good to go on my end!

File: grove/bernstein_vazirani/bernstein_vazirani.py (hunk 1 above)
Comment (joelin0) [inline grove/bernstein_vazirani/bernstein_vazirani.py]: Done (fyi, no tests actually called `run`)

File: grove/bernstein_vazirani/bernstein_vazirani.py (hunk 2 above)
Comment (joelin0) [inline grove/bernstein_vazirani/bernstein_vazirani.py]: Nope, changed to Connection.

**Comment by tarballs-are-good** [comment]
Awesome, thanks.

**Pull request merged**